ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -252,21 +252,25 @@ public void transformForBqSchema(Map<String, Object> parent, List<Field> bqField
       Map<String, Object> additionalProperties) {
 
     // Clean the key names.
-    ImmutableSet.copyOf(parent.keySet())
-        .forEach(key -> parent.put(getAndCacheBqName(key), parent.remove(key)));
+    final Map<String, String> jsonFieldNames = new HashMap<>();
+    ImmutableSet.copyOf(parent.keySet()).forEach(key -> {
+      String transformedKey = getAndCacheBqName(key);
+      jsonFieldNames.put(transformedKey, key);
+      parent.put(transformedKey, parent.remove(key));
```

Comment:
I think we should build additionalProperties while iterating through parent, both to only iterate once and to fix the issue from @acmiyaguchi's [comment](https://github.com/mozilla/gcp-ingestion/pull/713/files#r307911925) by using the original name in the first place.
```suggestion
      Object value = parent.remove(k);
      if (fieldNames.contains(transformedKey) {
        parent.put(transformedKey, value);
      } else if (additionalProperties != null) {
        additionalProperties.put(key, value);
      }
```

Reply:
I'm going to pull this into a single loop as suggested.